airbyte-integrations/connectors/source-stripe/unit_tests/integration/test_application_fees_refunds.py

```diff
@@ -0,0 +1,518 @@
+                    _as_dict(
+                        _refunds_response()
+                        .with_record(_a_refund())
+                    )
+                )
+            ).build(),
+        )
+        http_mocker.get(
+            _application_fees_request().with_starting_after("parent_id").with_created_gte(_A_START_DATE).with_created_lte(_NOW).with_limit(100).build(),
+            _application_fees_response()
+            .with_record(
+                _an_application_fee()
+                .with_field(
+                    _REFUNDS_FIELD,
+                    _as_dict(
+                        _refunds_response()
+                        .with_record(_a_refund())
+                    )
+                )
+            ).build(),
+        )
+
+        output = self._read(_config().with_start_date(_A_START_DATE))
+
+        assert len(output.records) == 2
+
+    @HttpMocker()
+    def test_given_parent_stream_without_refund_when_read_then_stream_is_unavailable(self, http_mocker: HttpMocker) -> None:
+        # events stream is not validated as application fees is validated first
+        http_mocker.get(
+            _application_fees_request().with_created_gte(_A_START_DATE).with_created_lte(_NOW).with_limit(100).build(),
+            _application_fees_response().build(),
+        )
+
+        output = self._read(_config().with_start_date(_A_START_DATE))
+
+        _assert_not_available(output)
+
+    @HttpMocker()
+    def test_given_slice_range_when_read_then_perform_multiple_requests(self, http_mocker: HttpMocker) -> None:
+        start_date = _NOW - timedelta(days=30)
+        slice_range = timedelta(days=20)
+        slice_datetime = start_date + slice_range
+
+        _given_events_availability_check(http_mocker)
+        http_mocker.get(
+            _application_fees_request().with_created_gte(start_date).with_created_lte(slice_datetime).with_limit(100).build(),
+            _application_fees_response().with_record(
+                _an_application_fee()
+                .with_field(_REFUNDS_FIELD, _as_dict(_refunds_response().with_record(_a_refund())))
+            ).build(),
+        )
+        http_mocker.get(
+            _application_fees_request().with_created_gte(slice_datetime + _AVOIDING_INCLUSIVE_BOUNDARIES).with_created_lte(_NOW).with_limit(100).build(),
+            _application_fees_response().with_record(
+                _an_application_fee()
+                .with_field(_REFUNDS_FIELD, _as_dict(_refunds_response().with_record(_a_refund())))
+            ).build(),
+        )
+
+        output = self._read(_config().with_start_date(start_date).with_slice_range_in_days(slice_range.days))
+
+        assert len(output.records) == 2
+
+    @HttpMocker()
+    def test_given_slice_range_and_refunds_pagination_when_read_then_do_not_slice_child(self, http_mocker: HttpMocker) -> None:
+        """
+        This means that if the user attempt to configure the slice range, it will only apply on the parent stream
```

Comment:
I don't know if it is worth an issue as I'm not sure why we allow for `slice_range`. Using `slice_range` allows for more or less slices but the requests are still limited to 100 entries as we always use `limit=100`. Hence, I'm not sure what is the value of slicing more often. The only value I see is that we would save the state more often in which case, it's fine that child are not being sliced.